rust/opendp/src/dom.rs

```diff
@@ -17,6 +17,9 @@ use crate::error::Fallible;
 pub struct AllDomain<T> {
     _marker: PhantomData<T>,
 }
+impl<T> Default for AllDomain<T> {
+    fn default() -> Self { AllDomain { _marker: PhantomData } }
```

Comment:
I think it's a little cleaner if these call `Self::new()`?

Reply:
Agreed, switched.